src/main/java/com/codeborne/selenide/commands/Click.java

```diff
@@ -4,87 +4,27 @@
 import com.codeborne.selenide.Command;
 import com.codeborne.selenide.Driver;
 import com.codeborne.selenide.SelenideElement;
-import com.codeborne.selenide.impl.FileContent;
 import com.codeborne.selenide.impl.WebElementSource;
 import org.openqa.selenium.WebElement;
 
 import javax.annotation.Nullable;
 import javax.annotation.ParametersAreNonnullByDefault;
 
-import static com.codeborne.selenide.commands.Util.firstOf;
-
 @ParametersAreNonnullByDefault
 public class Click implements Command<Void> {
-  private final FileContent jsSource = new FileContent("click.js");
 
   @Override
   @Nullable
   public Void execute(SelenideElement proxy, WebElementSource locator, @Nullable Object[] args) {
     Driver driver = locator.driver();
     WebElement webElement = locator.findAndAssertElementIsInteractable();
-
     if (args == null || args.length == 0) {
-      click(driver, webElement);
-    }
-    else if (args.length == 1) {
-      ClickOptions clickOptions = firstOf(args);
-      click(driver, webElement, clickOptions);
-    }
-    else if (args.length == 2) {
-      Integer offsetX = firstOf(args);
-      Integer offsetY = (Integer) args[1];
-      click(driver, webElement, offsetX, offsetY);
+      ClickOptions.clickUsingConfigMethod(driver, webElement);
+    } else if (args[0] instanceof Integer) {
+      ClickOptions.clickUsingConfigMethod(driver, webElement, (Integer) args[0], (Integer) args[1]);
+    } else {
+      Util.<ClickOptions>firstOf(args).click(driver, webElement);
```

Comment:
I guess it will fail when we call `void click(Duration timeout);`? If I am not mistaken, we need to ignore Duration inside Click command. Could you re-check it, please?

Reply:
Yes, I fixed it with new mechanism `OverloadedMethodCommandBuilder`. I'm sure we can reuse it for other commands with method overloading.